pageserver/src/tenant/storage_layer/batch_split_writer.rs

```
@@ -262,7 +274,13 @@ impl<'a> SplitImageLayerWriter<'a> {
             );
             self.start_key = key;
         }
-        self.inner.put_image(key, img, ctx).await
+        self.inner
+            .put_image(key, img, ctx)
+            .await
+            .map_err(|e| match e {
+                ImageLayerWriterError::Cancelled => anyhow::anyhow!("flush task cancelled"),
```

Propagate this error up instead of returning anyhow